File: project/src/nwr/core/types/ItfElement.java
/*
 *  "NorseWorld: Ragnarok", a roguelike game for PCs.
 *  Copyright (C) 2002-2008, 2014 by Serg V. Zhdanovskih (aka Alchemist).
 *
 *  This file is part of "NorseWorld: Ragnarok".
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package nwr.core.types;

import java.util.HashMap;
import nwr.engine.BaseScreen;

/**
 *
 * @author Serg V. Zhdanovskih
 */
public enum ItfElement
{
    id_Up(0, "Up.tga", BaseScreen.clWhite),
    id_Down(1, "Down.tga", BaseScreen.clWhite),
    id_Left(2, "Left.tga", BaseScreen.clWhite),
    id_Right(3, "Right.tga", BaseScreen.clWhite),
    id_Cursor(4, "Cursor.tga", BaseScreen.clWhite),
    id_Cursor_N(5, "Cursor_N.tga", BaseScreen.clWhite),
    id_Cursor_S(6, "Cursor_S.tga", BaseScreen.clWhite),
    id_Cursor_W(7, "Cursor_W.tga", BaseScreen.clWhite),
    id_Cursor_E(8, "Cursor_E.tga", BaseScreen.clWhite),
    id_Cursor_NW(9, "Cursor_NW.tga", BaseScreen.clWhite),
    id_Cursor_NE(10, "Cursor_NE.tga", BaseScreen.clWhite),
    id_Cursor_SW(11, "Cursor_SW.tga", BaseScreen.clWhite),
    id_Cursor_SE(12, "Cursor_SE.tga", BaseScreen.clWhite),
    id_FileDelete(13, "FileDelete.tga", BaseScreen.clNone),
    id_FileNum(14, "FileNum.tga", BaseScreen.clNone);


    public static final int id_First = 0;
    public static final int id_Last = 14;

    private final int intValue;
    public final String fName;
    public final int transColor;
    public int ImageIndex;

    private static HashMap<Integer, ItfElement> mappings;

    private static HashMap<Integer, ItfElement> getMappings()
    {
        synchronized (ItfElement.class) {
            if (mappings == null) {
                mappings = new HashMap<>();
            }
        }
        return mappings;
    }

    private ItfElement(int value, String fName, int transColor)
    {
        this.intValue = value;
        this.fName = fName;
        this.transColor = transColor;
        this.ImageIndex = -1;

        getMappings().put(value, this);
    }

    public int getValue()
    {
        return intValue;
    }

    public static ItfElement forValue(int value)
    {
        return getMappings().get(value);
    }
}

File: project/Core/Types/ItfElement.cs
/*
 *  "NorseWorld: Ragnarok", a roguelike game for PCs.
 *  Copyright (C) 2002-2008, 2014 by Serg V. Zhdanovskih.
 *
 *  This file is part of "NorseWorld: Ragnarok".
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace NWR.Core.Types
{
    public enum ItfElement
    {
        id_Up,
        id_Down,
        id_Left,
        id_Right,
        id_Cursor,
        id_Cursor_N,
        id_Cursor_S,
        id_Cursor_W,
        id_Cursor_E,
        id_Cursor_NW,
        id_Cursor_NE,
        id_Cursor_SW,
        id_Cursor_SE,
        id_FileDelete,
        id_FileNum,
        
        id_First = id_Up,
        id_Last = id_FileNum
    }
}
